pkg/testutil/sets.go

```diff
@@ -0,0 +1,62 @@
+package testutil
+
+type Set[K comparable] struct {
+	entities map[K]any
+}
+
+func NewSetFromMap[K comparable, V any](m map[K]V) *Set[K] {
```

Comment:
It'd be cool if these functions had comments. I haven't used any generics yet, so this is new for me.

Reply:
done!